File: ASPNET_MVC5/WinInvestmentTracker/Models/DEL/InvestmentNote.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;
using WinInvestmentTracker.Models.DEL.Interfaces;

namespace WinInvestmentTracker.Models.DEL
{
    public class InvestmentNote: IDbInvestmentEntity
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public int ID { get; set; }
        /// <summary>
        /// Title of the note
        /// </summary>
        public string Name { get; set; }
        /// <summary>
        /// Contents of the note
        /// </summary>
        public string Description { get; set; }
        /// <summary>
        /// All notes will be in relation to an investment
        /// </summary>
        public virtual Investment Investment { get; set; }        

    }
}

File: ASPNET_MVC5/angular/src/app/Models/InvestmentNote.ts
import { RisksLink } from './investment';
import { EntityTypes } from '../Utilities';
export interface InvestmentNote {
  id: number;
  name: string;
  description: string;
  owningEntityId: number;
  owningEntityType: EntityTypes;
}

/*
[
    {
        "id":17,
        "name":"note for investment#0",
        "description":"note for investment#0",
        "owningEntityId":1,
        "owningEntityType":0
    }
]
*/
